app/client/cli/debug.go

```diff
@@ -181,41 +179,25 @@ func initDebug(remoteCLIURL string) {
 		var err error
 		runtimeMgr := runtime.NewManagerFromFiles(defaultConfigPath, defaultGenesisPath, runtime.WithRandomPK())
 
-		consM, err := consensus.Create(runtimeMgr)
-		if err != nil {
-			logger.Global.Fatal().Err(err).Msg("Failed to create consensus module")
-		}
-		consensusMod = consM.(modules.ConsensusModule)
-
-		p2pM, err := p2p.Create(runtimeMgr)
-		if err != nil {
-			log.Fatalf("[ERROR] Failed to create p2p module: %v", err.Error())
-		}
-		p2pMod = p2pM.(modules.P2PModule)
+		// HACK: this is a temporary solution that guarantees backward compatibility while we implement peer discovery (#416).
+		validators = runtimeMgr.GetGenesis().Validators
 
-		// This telemetry module instance is a NOOP because the 'enable_telemetry' flag in the `cfg` above is set to false.
-		// Since this client mimics partial - networking only - functionality of a full node, some of the telemetry-related
-		// code paths are executed. To avoid those messages interfering with the telemetry data collected, a non-nil telemetry
-		// module that NOOPs (per the configs above) is injected.
-		telemetryM, err := telemetry.Create(runtimeMgr)
-		if err != nil {
-			logger.Global.Fatal().Err(err).Msg("Failed to create telemetry module")
-		}
-		telemetryMod := telemetryM.(modules.TelemetryModule)
+		debugAddressBookProvider := debugABP.NewDebugAddrBookProvider(
+			runtimeMgr.GetConfig().P2P,
+			debugABP.WithActorsByHeight(
+				map[int64][]*coreTypes.Actor{
+					debugABP.ALL_HEIGHTS: validators,
+				},
+			),
+		)
 
-		loggerM, err := logger.Create(runtimeMgr)
-		if err != nil {
-			logger.Global.Fatal().Err(err).Msg("Failed to create logger module")
-		}
-		loggerMod := loggerM.(modules.LoggerModule)
+		debugCurrentHeightProvider := debugCHP.NewDebugCurrentHeightProvider(0)
 
-		rpcM, err := rpc.Create(runtimeMgr)
+		p2pM, err := p2p.CreateWithProviders(runtimeMgr, debugAddressBookProvider, debugCurrentHeightProvider)
```

Comment:
Personally I am not a fan of mocks outside of tests. I prefer real, fit for purpose, simple implementations that scream "debug" so that the reader knows it is OK if it's not super clean code.

As P2P takes shape and we put in place real business logic, these guys are gonna be 🚮